packages/node-cli/src/commands/object/update/UpdateAction.js

```diff
@@ -47,26 +53,86 @@ module.exports = class UpdateAction extends BaseAction {
+
+				delete params[COMMAND_OPTIONS.INCLUDE_INSTANCES];
+			}
+
+			if (params[COMMAND_OPTIONS.SCRIPT_ID] === '') {
+				// If there are no more objects to update (all were custom records) we already return with only previous messages
+				return ActionResult.Builder.withData(totalData).build();
+			}
+
+			return await this._updateObjects(params, totalData);
 		} catch (error) {
 			return ActionResult.Builder.withErrors([error]).build();
 		}
 	}
+
+	async _updateObjects(params, totalData) {
+		const sdkParams = CommandUtils.extractCommandOptions(params, this._commandMetadata);
+
+		const executionContextForUpdate = SdkExecutionContext.Builder.forCommand(this._commandMetadata.sdkCommand)
+			.integration()
+			.addParams(sdkParams)
+			.build();
+
+		const operationResult = await executeWithSpinner({
+			action: this._sdkExecutor.execute(executionContextForUpdate),
+			message: NodeTranslationService.getMessage(MESSAGES.UPDATING_OBJECTS),
+		});
+		totalData = totalData.concat(operationResult.data);
+
+		return operationResult.status === STATUS.SUCCESS
+			? ActionResult.Builder.withData(totalData).withResultMessage(operationResult.resultMessage).withCommandParameters(sdkParams).build()
+			: ActionResult.Builder.withErrors(operationResult.errorMessages).withCommandParameters(sdkParams).build();
+	}
+
+	_isCustomRecordOrSegment(scriptid) {
+		return scriptid.startsWith(SCRIPT_ID_PREFIXES.CUSTOM_RECORD) || scriptid.startsWith(SCRIPT_ID_PREFIXES.CUSTOM_SEGMENT);
+	}
+
+	async _updateCustomRecordWithInstances(params, customRecordScriptIds) {
+		this._log.result(NodeTranslationService.getMessage(OUTPUT.UPDATED_CUSTOM_RECORDS));
+		const copiedParams = { ...params };
+		const dataResults = [];
+
+		for (const scriptId of customRecordScriptIds) {
+			const operationResult = await this._executeCommandUpdateCustomRecordWithInstances(copiedParams, scriptId);
+
+			if (operationResult.status === STATUS.ERROR) {
+				dataResults.push({ key: scriptId, message: operationResult.errorMessages, type: operationResult.status, includeinstances: true });
+				this._log.warning(NodeTranslationService.getMessage(ERRORS.CUSTOM_RECORD, scriptId, operationResult.errorMessages));
+			} else {
+				dataResults.push({ key: scriptId, message: operationResult.data, type: operationResult.status, includeinstances: true });
+				this._log.result(`${this._log.getPadding(1)}- ${scriptId}`);
+			}
+		}
+		return dataResults;
+	}
+
+	async _executeCommandUpdateCustomRecordWithInstances(params, scriptId) {
+		params[COMMAND_OPTIONS.SCRIPT_ID] = scriptId;
+		const executionContextForUpdate = SdkExecutionContext.Builder.forCommand(COMMAND_UPDATE_CUSTOM_RECORD_WITH_INSTANCES)
+			.integration()
+			.addParam(COMMAND_OPTIONS.AUTH_ID, params[COMMAND_OPTIONS.AUTH_ID])
+			.addParam(COMMAND_OPTIONS.PROJECT, params[COMMAND_OPTIONS.PROJECT])
+			.addParam(COMMAND_OPTIONS.SCRIPT_ID, scriptId)
+			.build();
+		const operationResult = await executeWithSpinner({
```

Comment:
You can return here instead of assigning it first to an operationResult variable.